src/events.js

```diff
@@ -1,68 +1,98 @@
+// @ts-check
 const e = require("./errors");
 const v = require("./validations");
 
+/**
+ * @typedef {import("..").ElectroEventListener} ElectroEventListener
+ * @typedef {import("..").ElectroEvent} ElectroEvent
+ */
+
+/**
+ * @see {@link https://electrodb.dev/en/reference/events-logging/ | Events and Logging} for more information.
+ * @class
+ */
 class EventManager {
+  /**
+   * @type {Array<ElectroEventListener>}
+   * */
+  #listeners;
+
+  /**
+   * Wraps the provided listener in order to safely invoke it.
+   * @static
+   * @template {(...args: any[]) => void} T
+   * @param {T} [listener] - The listener to wrap.
+   */
   static createSafeListener(listener) {
     if (listener === undefined) {
-      return undefined;
+      // no-op
+      return () => {};
```

Comment:
I could not get the filter to understand that undefined would have been removed, so changed it to a no-op and removed the filter entirely.